### PR TITLE
Enable kafkametricsreceiver.

### PR DESCRIPTION
**Pull request by dshomoye**
**Description:** This PR enables the `kafkametricsreceiver` by adding it to the `components.go` file.
Also:
- the default regex filter is updated to exclude internal kafka topics by default.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:**  Integration/e2e test is added, using a docker container that has zookeeper/kafka on it. This tests that at least one metric is returned (the number of brokers).

**Documentation:** README updated with default values and custom labels with additional sample config.

### CODE REVIEW

File: receiver/kafkametricsreceiver/README.md [hunk 1]
```diff
@@ -12,14 +12,17 @@ Required settings (no defaults):
...
 Optional Settings (with defaults):
 
 - `brokers` (default = localhost:9092): the list of brokers to read from.
-- `topic_match` (default = *): regex pattern of topics to filter for metrics collection.
-- `group_match` (default = *): regex pattern of consumer groups to filter on for metrics.
+- `topic_match` (default = ^[^_].*$): regex pattern of topics to filter on metrics collection. The default filter excludes internal topics (starting with `_`).
+- `group_match` (default = .*): regex pattern of consumer groups to filter on for metrics.
 - `client_id` (default = otel-metrics-receiver): consumer client id
 - `collection_interval` (default = 1m): frequency of metric collection/scraping.
+- `labels` (default none): list of additional labels to be applied to metrics. This should be in `key:value` format.
```
Comment (jrcamp) [inline receiver/kafkametricsreceiver/README.md:25]: Generally this kind of generic operation should be done in a processor, e.g. metricstransformprocessor.

File: receiver/kafkametricsreceiver/kafkametrics_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// Copyright  OpenTelemetry Authors
```
Comment (jrcamp) [inline receiver/kafkametricsreceiver/kafkametrics_e2e_test.go:1]: You'll want to add to `INTEGRATION_TEST_MODULES` in Makefile.

File: receiver/kafkametricsreceiver/kafkametrics_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	cfg.CollectionInterval = 5 * time.Second
+	consumer := new(consumertest.MetricsSink)
+	params := component.ReceiverCreateParams{Logger: zaptest.NewLogger(t)}
+
+	var receiver component.MetricsReceiver
+	var err error
+	t.Logf("waiting to connect to kafka...")
+	require.Eventuallyf(t,
+		func() bool {
+			receiver, err = f.CreateMetricsReceiver(context.Background(), params, cfg, consumer)
```
Comment (jrcamp) [inline receiver/kafkametricsreceiver/kafkametrics_e2e_test.go:73]: Does CreateMetricsReceiver actually require the server to be up? When can it fail?

File: receiver/kafkametricsreceiver/README.md (hunk 1 above)
Comment (dshomoye) [reply receiver/kafkametricsreceiver/README.md:25]: oh wow, I completely missed this. I will remove the custom labels then.

File: receiver/kafkametricsreceiver/kafkametrics_e2e_test.go (hunk 3 above)
Comment (dshomoye) [reply receiver/kafkametricsreceiver/kafkametrics_e2e_test.go:73]: `createMetricsReceiver` calls the `create` method for each scraper. This is where the `client` and/or `admin` from sarama is instantiated (the broker(s) are contacted at this point and will fail if they are not reachable which will cause the receiver to not be created).

Comment (jrcamp) [reply receiver/kafkametricsreceiver/kafkametrics_e2e_test.go:73]: Hmm generally Create* should be deterministic and not fail if network is down, etc. as it blocks pipeline creation. Would it be possible to delay connection until Start? Can do in a followup PR.

**Review by jrcamp — APPROVED** [review]

File: receiver/kafkametricsreceiver/kafkametrics_e2e_test.go (hunk 3 above)
Comment (dshomoye) [reply receiver/kafkametricsreceiver/kafkametrics_e2e_test.go:73]: ohhh I see, I'll have a follow up PR and will move the sarama initialization to the `Start` method of the scrapers then.

File: go.mod [hunk 4]
```diff
@@ -212,3 +213,5 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/processor/span
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter => ./exporter/googlecloudexporter
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/exporter/stackdriverexporter => ./exporter/stackdriverexporter
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkametricsreceiver => ./receiver/kafkametricsreceiver
```
Comment (bogdandrutu) [inline go.mod:217]: Please try to keep things sorted.

**Review by bogdandrutu — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (dshomoye) [reply go.mod:217]: the existing replace statements aren't sorted, that's why I added this to the last line. I assumed that was the pattern.
I'll move this under the rest of the receivers.

**Pull request merged**